src/black/trans.py

```diff
@@ -1243,6 +1243,41 @@ def more_splits_should_be_made() -> bool:
             last_line.comments = line.comments.copy()
             yield Ok(last_line)
 
+    def _get_nameescape_slices(self, string: str) -> List[Tuple[int, int]]:
+        """
+        Returns:
+            List of all ranges of @string which, if @string were to be split there,
+            would result in the splitting of an \\N{...} expression (which is NOT
+            allowed).
+        """
+        slices = []
+        backslash_count = 0
+        it = iter(enumerate(string))
+        for idx, c in it:
+            if not backslash_count:
+                if c == "\\":
+                    backslash_count += 1
+                continue
+            if c == "\\":
+                backslash_count += 1
+                continue
+            backslash_count = 0
+            if c != "N":
+                continue
```

Comment:
If I'm not mistaken, with escaped backslashes this would function incorrectly: `\\N{something}` would result in `count = 2` and still finding a slice. So how about something like:

```python
previous_was_backslash = False
for idx, c in it:
    if c == "\\":
        if not previous_was_backslash:
            previous_was_backslash = True
        else:
            previous_was_backslash = False    
        continue

    if not previous_was_backslash or c != "N":
        continue

    # Do the slice
```
Other potential names: `unescaped`, `previous_was_unescaped`, `previous_backslash_was_unescaped` (lol maybe not)

Reply:
`backslash_count` was supposed to be used for escapes but looks I completely forgot to actually put such handling in...
I'll look into fixing this and adding more tests to cover it later today.

Reply:
Yeah I think testing for an odd number of backslashes works just as well if you were going for that!

Reply:
Actually now I see that my version wouldn't reset if some other character was after a backslash, so just keep yours I think 😅

Reply:
Escapes should work now, I decided to go for a tri-bool.